libraries/preferences/api/src/main/kotlin/io/element/android/libraries/preferences/api/store/EnableNativeSlidingSyncUseCase.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.libraries.preferences.api.store
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class EnableNativeSlidingSyncUseCase @Inject constructor(
+    private val appPreferencesStore: AppPreferencesStore,
+    private val appCoroutineScope: CoroutineScope,
+) {
+    operator fun invoke() {
+        appCoroutineScope.launch {
+            appPreferencesStore.setSimplifiedSlidingSyncEnabled(true)
```

Comment:
There is maybe an issue if the user decides to login using another homeserver... I agree this is an edge case and this is temporary code, but with the current codebase, but we should maybe have a setting per server...

Reply:
Since this forces you to log out, I don't think most users will try it and then change their mind and log into another home server. Besides, changing this would required quite a few logic changes and maybe UI ones for the developer options toggle, and it seems like an overkill for something like this.

What we could do, maybe, is that when this is enabled, the 1st login attempt will be tried using a native sliding sync config, then if it fails with a not supported error the 2nd one will use the previous proxy one as a fallback. WDYT? Maybe I can add this in a separate PR.

Reply:
OK, let's not do anything then.
Also there is another (limited) risk of user seeing the banner and decide to logout from the settings. In this case, there will be no usage of the SSS and the banner will reappear.